src/lib/core/datetime.c

```diff
@@ -134,6 +135,57 @@ datetime_to_string(const struct datetime *date, char *buf, ssize_t len)
 	return sz;
 }
 
+size_t
+datetime_parse_full(struct datetime *date, const char *str, size_t len,
+		    int32_t offset)
+{
+	size_t n;
+	dt_t dt;
+	const char *svp = str;
+	char c;
+	int sec_of_day = 0, nanosecond = 0;
+
+	n = dt_parse_iso_date(str, len, &dt);
+	if (n == 0)
+		return 0;
+	if (n == len)
+		goto exit;
+
+	c = str[n++];
+	if (c != 'T' && c != 't' && c != ' ')
+		return 0;
+
+	str += n;
+	len -= n;
+
+	n = dt_parse_iso_time(str, len, &sec_of_day, &nanosecond);
+	if (n == 0)
+		return 0;
+	if (n == len)
+		goto exit;
+
+	if (str[n] == ' ')
+		n++;
+
+	str += n;
+	len -= n;
+
+	n = dt_parse_iso_zone_lenient(str, len, &offset);
+	if (n == 0 || n != len)
+		return 0;
+	str += n;
+
+exit:
+	date->epoch =
+		((int64_t)dt_rdn(dt) - DT_EPOCH_1970_OFFSET) * SECS_PER_DAY +
+		sec_of_day - offset * 60;
+	date->nsec = nanosecond;
+	date->tzoffset = offset;
+	date->tzindex = 0;
+
+	return (str - svp);
```

Comment:
One place with strange braces around the return value left :)

Reply:
Fixed